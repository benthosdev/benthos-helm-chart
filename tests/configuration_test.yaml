suite: configuration tests
templates:
  - configmap.yaml
tests:
  - it: should set default configuration
    asserts:
      - matchSnapshot:
          path: data

  - it: should set http options
    set:
      http:
        address: 127.0.0.0:4095
        cert_file: "/my/cert"
        key_file: "/my/cert.key"
        cors:
          enabled: true
          allowed_origins: []
        basic_auth:
          enabled: true
          username: "test"
          password_hash: "test"
          algorithm: "sha256"
          salt: "test"
    asserts:
      - matchSnapshot:
          path: data

  - it: should set logging options
    set:
      logger:
        level: INFO
        format: logfmt
        add_timestamp: false
        static_fields:
          "@service": benthos
    asserts:
      - matchSnapshot:
          path: data

  - it: should set tracing options
    set:
      tracing:
        jaeger:
          agent_address: localhost:6831
          sampler_type: const
          sampler_param: 1
    asserts:
      - matchSnapshot:
          path: data

  - it: should set metrics options
    set:
      metrics:
        statsd:
          address: localhost:8125
          flush_period: 100ms
    asserts:
      - matchSnapshot:
          path: data

  - it: can set global options with streams mode
    set:
      streams:
        enabled: true
        streamsConfigMap: my-config-map
      http:
        address: 127.0.0.0:4095
        cert_file: "/my/cert"
        key_file: "/my/cert.key"
        cors:
          enabled: true
          allowed_origins: []
        basic_auth:
          enabled: true
          username: "test"
          password_hash: "test"
          algorithm: "sha256"
          salt: "test"
      logger:
        level: INFO
        format: logfmt
        add_timestamp: false
        static_fields:
          "@service": benthos
      tracing:
        jaeger:
          agent_address: localhost:6831
          sampler_type: const
          sampler_param: 1
      metrics:
        statsd:
          address: localhost:8125
          flush_period: 100ms
    asserts:
      - matchSnapshot:
          path: data
