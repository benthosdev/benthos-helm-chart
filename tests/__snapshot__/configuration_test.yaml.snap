can set global options with streams mode:
  1: |
    benthos.yaml: |-
      logger:
        add_timestamp: false
        format: logfmt
        level: INFO
        static_fields:
          '@service': benthos
      metrics:
        statsd:
          address: localhost:8125
          flush_period: 100ms
      tracer:
        jaeger:
          agent_address: localhost:6831
          sampler_param: 1
          sampler_type: const
      http:
        address: 127.0.0.0:4095
        basic_auth:
          algorithm: sha256
          enabled: true
          password_hash: test
          salt: test
          username: test
        cert_file: /my/cert
        cors:
          allowed_origins: []
          enabled: true
        debug_endpoints: false
        enabled: true
        key_file: /my/cert.key
        root_path: /benthos
should set default configuration:
  1: |
    benthos.yaml: |-
      http:
        address: 0.0.0.0:4195
        cors:
          enabled: false
        debug_endpoints: false
        enabled: true
        root_path: /benthos
      input:
        label: "no_config_in"
        generate:
          mapping: root = "This Benthos instance is unconfigured!"
          interval: 1m
      output:
        label: "no_config_out"
        stdout:
          codec: lines
should set http options:
  1: |
    benthos.yaml: |-
      http:
        address: 127.0.0.0:4095
        basic_auth:
          algorithm: sha256
          enabled: true
          password_hash: test
          salt: test
          username: test
        cert_file: /my/cert
        cors:
          allowed_origins: []
          enabled: true
        debug_endpoints: false
        enabled: true
        key_file: /my/cert.key
        root_path: /benthos
      input:
        label: "no_config_in"
        generate:
          mapping: root = "This Benthos instance is unconfigured!"
          interval: 1m
      output:
        label: "no_config_out"
        stdout:
          codec: lines
should set logging options:
  1: |
    benthos.yaml: |-
      logger:
        add_timestamp: false
        format: logfmt
        level: INFO
        static_fields:
          '@service': benthos
      http:
        address: 0.0.0.0:4195
        cors:
          enabled: false
        debug_endpoints: false
        enabled: true
        root_path: /benthos
      input:
        label: "no_config_in"
        generate:
          mapping: root = "This Benthos instance is unconfigured!"
          interval: 1m
      output:
        label: "no_config_out"
        stdout:
          codec: lines
should set metrics options:
  1: |
    benthos.yaml: |-
      metrics:
        statsd:
          address: localhost:8125
          flush_period: 100ms
      http:
        address: 0.0.0.0:4195
        cors:
          enabled: false
        debug_endpoints: false
        enabled: true
        root_path: /benthos
      input:
        label: "no_config_in"
        generate:
          mapping: root = "This Benthos instance is unconfigured!"
          interval: 1m
      output:
        label: "no_config_out"
        stdout:
          codec: lines
should set tracing options:
  1: |
    benthos.yaml: |-
      tracer:
        jaeger:
          agent_address: localhost:6831
          sampler_param: 1
          sampler_type: const
      http:
        address: 0.0.0.0:4195
        cors:
          enabled: false
        debug_endpoints: false
        enabled: true
        root_path: /benthos
      input:
        label: "no_config_in"
        generate:
          mapping: root = "This Benthos instance is unconfigured!"
          interval: 1m
      output:
        label: "no_config_out"
        stdout:
          codec: lines
